drv/lpc55-update-server/src/main.rs

```diff
@@ -246,15 +239,20 @@ fn validate_header_block(
         return Ok(());
     }
 
+    // This part aliases flash in two positions that differ in bit 28. To allow
+    // for either position to be used in new images, we clear bit 28 in all of
+    // the numbers used for comparison below, by ANDing them with this mask:
+    const ADDRMASK: u32 = !(1 << 28);
+
     let reset_vector = u32::from_le_bytes(
         block[RESET_VECTOR_OFFSET..][..4].try_into().unwrap_lite(),
-    );
-    let a_base = unsafe { &__IMAGE_A_BASE } as *const u32 as u32;
-    let b_base = unsafe { &__IMAGE_B_BASE } as *const u32 as u32;
-    let stage0_base = unsafe { &__IMAGE_STAGE0_BASE } as *const u32 as u32;
-    let a_end = unsafe { &__IMAGE_A_END } as *const u32 as u32;
-    let b_end = unsafe { &__IMAGE_B_END } as *const u32 as u32;
-    let stage0_end = unsafe { &__IMAGE_STAGE0_END } as *const u32 as u32;
+    ) & ADDRMASK;
+    let a_base = unsafe { &__IMAGE_A_BASE } as *const u32 as u32 & ADDRMASK;
```

Comment:
Nit: I find it a bit cleaner to write this as
```suggestion
    let a_base = unsafe { __IMAGE_A_BASE.as_ptr() } as u32 & ADDRMASK;
```

Reply:
Yeah, so do I -- this is copy-pasted code. I will clean this up.

Reply:
(In the original these fields were all `: ()` so `.as_ptr()` wasn't available, I changed them to zero-length arrays)